lib/win_check.rb

```diff
@@ -0,0 +1,52 @@
+class WinCheck
+  attr_reader :board
+
+  def initialize(board)
+    @board = board
+  end
+
+    #The nested ranges below generate every coordinate associated with a cardinal and ordinal direciton. 
+    #for example (-1, -1) refers to one column to the left, and down one position, i.e. 'South-West', 'Diagonally down and to the left' etc.
+    #The function then recusively calls "check" in that direction until it encounters
+    #a position with a symbol that does not match the original piece.
+
+  def check_all(piece)
+    counts = []
+    #x_coord modifier
+    (-1..1).each do |x_mod|
+      #y_coord modifier
+      (-1..1).each do |y_mod|
+        
+        check_count = 0
+        #This line skips (0,0) which would check the piece against itself, infinitely.
+        next if x_mod == 0 && y_mod == 0
+        
+        piece_to_test = @board.fetch_piece((piece.x_pos + x_mod), (piece.y_pos + y_mod))
+        
+        count = check(piece, piece_to_test, x_mod, y_mod, check_count)
+        #Reverse check. For any direction, we need to also check in the opposite direction and add it to our total checks,
+        #in case our piece is in the middle of a line of connected pieces. 
+        #this is accomplish by inverting the sign of our coordinate modifiers.
+        x_mod, y_mod = -x_mod, -y_mod
+        piece_to_test = @board.fetch_piece((piece.x_pos + x_mod), (piece.y_pos + y_mod))
+        reverse_count = check(piece, piece_to_test, x_mod, y_mod, check_count )
+        total = count + reverse_count
+        counts << total
+      end
+    end
+    
+    counts.any?{|count| count >= 3}
```

Comment:
the forth piece will be the piece that we have placed and are checking against the other three pieces, correct?